<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Codes</title>
    <link rel="stylesheet" href="../../styles/program.css">
    <link rel="icon" href="../../images/icons/favicon.png" type="image/x-icon">
</head>

<body>
    <!-- Navigation Bar -->
    <div class="navbar">
        <div class="logo">
            <img src="../../images/logo.png" alt="Logo">
        </div>
        <div class="nav-links">
            <a href="../../index.html">Home</a>
            <div class="dropdown">
                <button class="dropbtn">Notes &#9662</button>
                <div class="dropdown-content">
                    <a href="../../notes/first.html">First</a>
                    <a href="../../notes/second.html">Second</a>
                    <a href="../../notes/third.html">Third</a>
                    <a href="../../notes/fourth.html">Fourth</a>
                </div>
            </div>
            <a href="../../labcodes.html">Lab Codes</a>
            <a href="../../about.html">About</a>
        </div>
    </div>

    <main>
        <div class="content-container">
            <h1>Finally- exception handling</h1>
            <div class="code-box">
                <p>
                    import java.util.Scanner;<br>
                            public class nineteen {<br>
                            public static void main(String[] args) {<br>
                            Scanner scanner = new Scanner(System.in);<br>
                            try {<br>
                            System.out.print("Enter a number: ");<br>
                            int num = scanner.nextInt();<br>
                            int result = divideByTwo(num);<br>
                            System.out.println("Result of dividing by 2: " + result);<br>
                            }<br>
                            catch (ArithmeticException e) {<br>
                            System.out.println("Exception: " + e.getMessage());<br>
                            }<br>
                            finally {<br>
                            System.out.println("Finally block is executed, regardless of whether an exception occurred
                            or not.");<br>
                            scanner.close(); // Close the Scanner to release resources<br>
                            }<br>
                            }<br>
                            public static int divideByTwo(int num) {<br>
                            if (num % 2 != 0) {<br>
                            throw new ArithmeticException("Number must be even for this operation.");<br>
                            }<br>
                            return num / 2;<br>
                            }<br>
                            }<br>
                </p>
            </div>
        </div>
    </main>
</body>

</html>